### ISSUE TITLE
Predictable Scale: Part I — Optimal Hyperparameter Scaling Law in Large Language Model Pretraining

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#382]
## Links
https://arxiv.org/abs/2503.04715v1

## Authors
[Houyi Li](https://arxiv.org/search/cs?searchtype=author&query=Li,+H), [Wenzhen Zheng](https://arxiv.org/search/cs?searchtype=author&query=Zheng,+W), [Qiufeng Wang](https://arxiv.org/search/cs?searchtype=author&query=Wang,+Q), [Hanshan Zhang](https://arxiv.org/search/cs?searchtype=author&query=Zhang,+H), [Zili Wang](https://arxiv.org/search/cs?searchtype=author&query=Wang,+Z), [Shijie Xuyang](https://arxiv.org/search/cs?searchtype=author&query=Xuyang,+S), [Yuantao Fan](https://arxiv.org/search/cs?searchtype=author&query=Fan,+Y), [Shuigeng Zhou](https://arxiv.org/search/cs?searchtype=author&query=Zhou,+S), [Xiangyu Zhang](https://arxiv.org/search/cs?searchtype=author&query=Zhang,+X), [Daxin Jiang](https://arxiv.org/search/cs?searchtype=author&query=Jiang,+D)

## TLDR
- batch size は固定
- LLM における LR を BS ごとにいっぱい調べた
- 論文で採用したのは**「固定最終学習率（Fixed Final Learning Rate）」**と呼ばれるスケジュールです。
  - 具体的には以下のような設定となっています：
    - 初期ウォームアップ（linear warmup）：最初の2000ステップにわたって線形に学習率を上昇させます。
    - コサイン減衰（cosine decay）：ウォームアップ終了後は、最終的な最小学習率（min_lr）に向かって、コサイン曲線で徐々に学習率を減衰させます。

## 主な貢献（Contributions）
本論文は、大規模言語モデル（LLM）の事前学習において最適な学習率（Learning Rate, LR）とバッチサイズ（Batch Size, BS）のスケーリング法則を示し、その普遍性・一般性を明らかにした研究である。特に、モデル規模やデータサイズに応じて最適なハイパーパラメータを予測可能な「Step Law」を導出し、実証的な検証を通じて従来研究との差異や新規性を明確にしている。

<img width="297" height="309" alt="Image" src="https://github.com/user-attachments/assets/c140147c-2c73-4f16-b038-eef4da44cf4b" />

## 論文の新規性・従来研究との差異
本研究の新規性は以下のようにまとめられる。

### 凸性（Convexity）の発見

本研究が初めて明らかにしたのは、モデルのパラメータ数とデータサイズを固定した場合の損失関数（loss）のランドスケープが、LRとBSに対して凸型（convex）を示すことである。
この凸性は最適なハイパーパラメータの領域が一定範囲に収束することを示し、実務的にも重要な示唆を与える。

<img width="939" height="485" alt="Image" src="https://github.com/user-attachments/assets/851a584f-cb9a-4253-9172-d069a109d41c" />

<img width="590" height="264" alt="Image" src="https://github.com/user-attachments/assets/86eb95f0-d191-4d7f-8c6b-8a182937f21d" />

### ユニバーサルなハイパーパラメータスケーリング法則の導出

従来の研究（Kaplanら, 2020; DeepSeekら, 2024; Porianら, 2024など）は主にモデルサイズや特定の条件下でのみスケーリング法則を示していたが、本研究はモデル構造（密なTransformerと疎なMoEモデル）やデータ分布の違いを超えた普遍的でロバストなスケーリング法則（Step Law）を導出した。

具体的には、最適な学習率はモデルのパラメータ数（N）とデータセットサイズ（D）の両方に依存し、バッチサイズは主にデータセットサイズに依存することを明らかにした。

Step Lawの具体的な式は以下の通りである:

<img width="634" height="256" alt="Image" src="https://github.com/user-attachments/assets/ea8a7998-4cfc-49c8-915e-d0b077cbe024" />

### 疎なモデル（MoEモデル）における汎用性の検証

密なTransformerだけでなく、パラメータの一部のみを活性化する疎なMoE（Mixture-of-Experts）モデルにもStep Lawが適用可能であることを初めて示した。
疎密度の違いによらず、高い精度で最適なハイパーパラメータを予測できることを検証した。

### 多様なデータ分布に対するロバスト性の実証

英語以外の多言語（中国語）やコード主体のデータセットなど、様々なデータ分布条件でもStep Lawが安定して最適な設定を予測できることを示し、従来の研究に対する明確な優位性を示した。

## 貢献を実証するための実験アプローチ
本研究は貢献を実証するため、以下のような大規模で包括的な実験を実施した。

### 大規模な実験設定

総計で3700以上の異なる条件下（密・疎モデル、多様なモデルサイズ・データサイズ・データ分布）で、モデルをスクラッチから事前学習し、実証的に最適なハイパーパラメータを特定した。
約1,000,000 GPU時間（NVIDIA H800）という莫大な計算リソースを投入し、合計100兆トークン規模のデータを使用して、信頼性の高い実験データを得た。

### 従来手法との比較・評価

Kaplan法、DeepSeek法、Porian法、Microsoft法など既存のスケーリング手法と比較実験を行い、Step Lawが最も正確に最適点を予測することを明確に示した（相対誤差0.07%）。
従来法は主に学習率（LR）のみに焦点を当てていたが、本研究はバッチサイズ（BS）も含め同時にモデル化しているため、より安定した予測性能を実現した。

### 凸性の実証的な検証

損失ランドスケープを精密に測定し、学習率とバッチサイズの2次元空間上での凸性を実証した（論文のFig. 2およびFig. 3参照）。

また、最適なハイパーパラメータ領域が安定した「プラトー」を持つことを示し、実務上の許容範囲を明確に定義した

<img width="306" height="440" alt="Image" src="https://github.com/user-attachments/assets/0ce4fa71-a2ef-4012-a39c-136096f2fbdb" />
